libs/gl-cli/src/util.rs

```diff
@@ -0,0 +1,91 @@
+use bip39::{self, Language, Mnemonic};
+use dirs;
+use gl_client::bitcoin::secp256k1::rand::{self, RngCore};
+use gl_client::credentials;
+use std::path::PathBuf;
+use std::{
+    fs::{self, File},
+    io::Write,
+    path::Path,
+};
+use thiserror;
+
+pub const SEED_FILE_NAME: &str = "hsm_secret";
+pub const CREDENTIALS_FILE_NAME: &str = "credentials.gfs";
+pub const DEFAULT_GREENLIGHT_DIR: &str = "greenlight";
+
+// -- Seed section
+
+pub fn generate_seed() -> Result<Vec<u8>> {
+    let mut entropy = [0u8; 32];
+    let mut rng = rand::thread_rng();
+    rng.fill_bytes(&mut entropy);
+
+    let mnemonic = Mnemonic::from_entropy_in(Language::English, &entropy)?;
+
+    // Return the first 32 bytes of the seed generated from the mnemonic
+    Ok(mnemonic.to_seed("")[0..32].to_vec())
+}
+
+pub fn read_seed(file_path: impl AsRef<Path>) -> Option<Vec<u8>> {
+    fs::read(file_path).ok()
+}
+
+pub fn write_seed(file_path: impl AsRef<Path>, seed: impl AsRef<[u8]>) -> Result<()> {
+    let file_path = file_path.as_ref();
+    if let Some(parent) = file_path.parent() {
+        fs::create_dir_all(parent)?;
+    }
+
+    let mut file = File::create(file_path)?;
+    file.write_all(seed.as_ref())?;
```

Comment:
You might want to `file.flush()?;` as well, just to ensure the seed is written fully.

Reply:
Alternatively `file.sync_data()?` could also be useful to flush.